### DISCUSSION TITLE
Is there a way to center a node vertically without changing the x position?

### DISCUSSION THREAD

**Discussion by ryosua** [xyflow/xyflow#2448] [Q&A]
I'm trying to center the flow vertically on a node when it is clicked. I see the example - https://reactflow.dev/docs/examples/misc/use-react-flow-hook/ which uses `setCenter`, but I do not want to change the x position. Is there a value I can pass to only change the center in the y direction?

**Comment by bcakmakoglu** [comment]
Yeah you can use the `transform` value from the store api.

```js
  const store = useStoreApi();
  const { zoomIn, zoomOut, setViewport } = useReactFlow();

 const focusNode = () => {
    const { nodeInternals, transform, domNode } = store.getState();
    const nodes = Array.from(nodeInternals).map(([, node]) => node);

    if (nodes.length > 0) {
      const node = nodes[0];

      setViewport(
        {
          x: transform[0],
          y: node.position.y + domNode.offsetHeight / 2,
          zoom: transform[2]
        },
        { duration: 1000 }
      );
    }
  };
```

[Here's a sandbox with a demo.](https://codesandbox.io/s/sandpack-project-forked-liekrq?file=/Sidebar.js:116-206)
